File: scm-csharp/scheme/primitives/PrimitiveProcessAliveQ.cs
namespace scheme;

public class PrimitiveProcessAliveQ : Primitive
{
    public override string Name() => "process-alive?";

    public override string Info() =>
        "Syntax: (process-alive? handle)\n" +
        "Library: (scm system)\n" +
        "Description: Returns #t if the process started by start-program is still " +
        "running, #f if it has exited.\n" +
        "Example:\n" +
        "  (process-alive? p) => #t";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        SchemeProcess sp = (SchemeProcess) Value.AsNativeValue(arguments[0]).value;
        return sp.process.HasExited ? Value.F : Value.T;
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveProcessAliveQ.java
package scheme.primitives;
import scheme.*;

public class PrimitiveProcessAliveQ extends Primitive {
    @Override
    public String name() { return "process-alive?"; }

    @Override
    public String info() {
        return "Syntax: (process-alive? handle)\n" +
               "Library: (scm system)\n" +
               "Description: Returns #t if the process started by start-program is still " +
               "running, #f if it has exited.\n" +
               "Example:\n" +
               "  (process-alive? p) => #t";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        SchemeProcess sp = (SchemeProcess) Value.asNativeValue(arguments[0]).value;
        return sp.process.isAlive() ? Value.T : Value.F;
    }
}
